File: [0002][add-two-numbers]/AddTwoNumbers.py
# Definition for singly-linked list.
class ListNode:
     def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        res = currentNode = ListNode(-1)
        carry = 0
        while l1 or l2 or carry:
            # use and to check whether the node pointer is none, if it is none, use 0 as its value on this bit.
            value = ((l1 and l1.val) or 0) + ((l2 and l2.val) or 0) + carry
            carry = int(value/10)
            currentNode.next = ListNode(value%10)
            l1 = l1 and l1.next
            l2 = l2 and l2.next
            currentNode = currentNode.next
        return res.next    

File: [0002][add-two-numbers]/AddTwoNumbers.c
//Definition for singly-linked list.
/* struct ListNode {
     int val;
    struct ListNode *next;
 }; */

struct ListNode *addTwoNumbers(struct ListNode *l1, struct ListNode *l2)
{
    int carry_num = 0;
    struct ListNode *currentNode = NULL;
    currentNode = malloc(sizeof(*currentNode));
    struct ListNode *res = currentNode;
    while (l1 != NULL || l2 != NULL || carry_num)
    {
        int sum, value_l1, value_l2;
        value_l1 = (l1 != NULL) ? l1->val : 0;
        value_l2 = (l2 != NULL) ? l2->val : 0;
        sum = value_l1 + value_l2 + carry_num;
        carry_num = sum / 10;
        currentNode->next = malloc(sizeof(*currentNode));
        currentNode->next->val = sum % 10;        
        currentNode->next->next = NULL;
        l1 = l1 != NULL ? l1->next : NULL;
        l2 = l2 != NULL ? l2->next : NULL;
        currentNode = currentNode->next;
    }
    return res->next;
}